### PR TITLE
ci: setup travis ci to automatically run test suite for each PR & commit

### PR DESCRIPTION
**Pull request by ChristianMurphy**
This will automatically run `yarn test` for each incoming commit and PR, and report the status back to GitHub.

https://travis-ci.com

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+  - 8
+  - 10
+  - 12
```
Comment (ChristianMurphy) [inline .travis.yml]: `iltorb` may fail during install phase

Comment (ChristianMurphy) [reply .travis.yml]: :notebook: this will be resolved by #594

**Comment by ChristianMurphy** [comment]
It appears Circle CI may also be setup, but is not reporting status.
ref #449 #527 #538 

What was the issue encountered that caused the migration away from Travis?
Any objections to bringing it back?

**Comment by wooorm** [comment]
I don't remember the issue, but there was something going on 🤷‍♂️ If it's green now, I'm personally up for bringing it back!

**Comment by johno** [comment]
Hmmm, the switch to CircleCI was intentional but I have no idea why it [stopped reporting 🤔 ](https://circleci.com/gh/mdx-js/mdx).

**Comment by ChristianMurphy** [comment]
digging into the circle CI config it looks like https://github.com/mdx-js/mdx/blob/deff36bebfedb3a9de0a0575ee9a1b55b9b8aa18/.circleci/config.yml#L14-L17 is limiting it to only running on `master` branch.

**Comment by ChristianMurphy** [comment]
> the switch to CircleCI was intentional

Okay, I'll repeat my question from before:

What was the issue encountered that caused the migration away from Travis?
Any objections to bringing it back?

**Comment by johno** [comment]
> What was the issue encountered that caused the migration away from Travis?

I was mostly worried about the Travis acquisition and downtime. Funnily enough, we subsequently saw lots of CircleCI downtime after moving.

> Any objections to bringing it back?

Nope. To me CI is CI so whatever works!

**Pull request merged**